src/components/TankBar/TankBar.js

```diff
@@ -0,0 +1,326 @@
+  font-size: ${fontSizes.tiny}px;
+  line-height: ${fontSizes.tiny}px;
+  color: ${props => props.color ? props.color : baseColors.slateBlack};
+`;
+
+// const ChevronWrapper = styled.View`
+//   flex-direction: column;
+//   align-items: center;
+//   justify-content: center;
+//   margin-left: 4px;
+//   margin-top: -2px;
+// `;
+//
+// const SelectorChevron = styled(Icon)`
+//   font-size: 5px;
+//   color: ${baseColors.slateBlack};
+// `;
+
+export default class TankBar extends React.Component<Props, State> {
+  static defaultProps = {
+    circle: false,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      progressAnimated: new Animated.Value(0),
+      barWidth: 0,
+      labelWidth: 0,
+      labelTransform: 0,
+      sideButtonWidth: 0,
+      didFirstAnimation: false,
+    };
+  }
+
+  componentDidMount() {
+    this.state.progressAnimated.addListener(({ value }) => this.updateLabelTransform(value));
+  }
+
+  updateLabelTransform = (value: number) => {
+    const { labelWidth, barWidth, sideButtonWidth } = this.state;
+    const barWidthPixels = barWidth * (value / 100);
+    const halfLabelWidth = labelWidth / 2;
+    let labelTransform = -(labelWidth / 2);
+    if (barWidthPixels <= halfLabelWidth) {
+      labelTransform = -barWidthPixels;
+    } else if (sideButtonWidth <= halfLabelWidth) {
+      labelTransform = -(halfLabelWidth - (sideButtonWidth - halfLabelWidth));
+    }
+    this.setState({ labelTransform });
+  };
+
+  firstAnimate = () => {
+    this.setState({ didFirstAnimation: true });
+    const { maxValue, currentValue } = this.props;
+    this.handleStatusValue(maxValue, currentValue);
+  };
+
+  handleStatusValue = (full: number, current: number) => {
+    let currentStatus = Math.floor((current * 100) / full);
+    if (current > full) currentStatus = 100;
+    this.animateProgress(currentStatus);
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    const { maxValue, currentValue } = this.props;
+    if (prevProps.currentValue !== currentValue) {
+      this.handleStatusValue(maxValue, currentValue);
+    }
+  }
+
+  animateProgress = (newProgress: number) => {
+    Animated.spring(this.state.progressAnimated, {
+      toValue: newProgress,
+      duration: 400,
+      bounciness: 2,
+    }).start();
+  };
+
+  render() {
+    const {
+      progressAnimated,
+      barWidth,
+      sideButtonWidth,
+      labelTransform,
+      didFirstAnimation,
+    } = this.state;
+    const {
+      maxValue,
+      currentValue,
+      currentValueFormatted,
+      topupAction,
+      topUpLoading,
+    } = this.props;
+
+    return (
+      <Wrapper>
+        <Row style={{ marginBottom: spacing.large, marginTop: spacing.small }}>
+          <Title>PLR Tank</Title>
+        </Row>
+        <Row style={{ alignItems: 'flex-start' }}>
+          <ProgressBarSide>
+            <ProgressBarWrapper
+              onLayout={(e) => {
+                const { width } = e.nativeEvent.layout;
```

Comment:
Will crash if `e` or `e.nativeEvent` or `e.nativeEvent.layout` is undefined.

Reply:
Will check and update in upcoming commit. Thanks! :)